### PR TITLE
Add squeeze to ATen XLA tensor

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/ops/squeeze.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+namespace torch_xla {
+namespace ir {
+namespace ops {
+namespace {
+
+xla::Shape NodeOutputShape(const Value& input, int dim) {
+  auto lower_for_shape_fn =
+      [dim](tensorflow::gtl::ArraySlice<const xla::XlaOp> operands)
+      -> xla::XlaOp {
+    XLA_CHECK_EQ(operands.size(), 1);
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/squeeze.cpp]: This can use the single function like we talked yesterday.

File: torch_xla/csrc/tensor.cpp [hunk 2]
```diff
@@ -64,6 +65,20 @@ void SetMulti(std::vector<XLATensor>* dest_tuple,
   }
 }
 
+// Get the canonical dimension index in the [0, rank) interval. Negative indices
+// are interpreted as follows: -1 is rank-1, -2 is rank-2 etc.
+int GetCanonicalDimensionIndex(int dim, int rank) {
+  int min_shape_dim = -rank;
+  int max_shape_dim = rank - 1;
+  XLA_CHECK(min_shape_dim <= dim && dim <= max_shape_dim) << absl::StrCat(
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor.cpp]: Why StrCat() when you can stream?

File: torch_xla/csrc/ops/squeeze.cpp (hunk 1 above)
Comment (asuhan) [reply torch_xla/csrc/ops/squeeze.cpp]: Done.

File: torch_xla/csrc/tensor.cpp (hunk 2 above)
Comment (asuhan) [reply torch_xla/csrc/tensor.cpp]: Done.

**Pull request merged**